sdk/src/main/java/io/opentelemetry/sdk/metrics/aggregator/DoubleSummaryAggregator.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright 2020, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.opentelemetry.sdk.metrics.aggregator;
+
+import com.google.errorprone.annotations.concurrent.GuardedBy;
+import io.opentelemetry.sdk.metrics.data.MetricData.DoubleSummaryPoint;
+import io.opentelemetry.sdk.metrics.data.MetricData.Point;
+import java.util.Map;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import javax.annotation.Nullable;
+import javax.annotation.concurrent.ThreadSafe;
+
+@ThreadSafe
+public class DoubleSummaryAggregator extends AbstractAggregator {
+  private static final AggregatorFactory AGGREGATOR_FACTORY =
+      new AggregatorFactory() {
+        @Override
+        public Aggregator getAggregator() {
+          return new DoubleSummaryAggregator();
+        }
+      };
+
+  // The current value. This controls its own internal thread-safety via method access. Don't
+  // try to use its fields directly.
+  private final DoubleSummary current = new DoubleSummary();
+
+  public static AggregatorFactory getFactory() {
+    return AGGREGATOR_FACTORY;
+  }
+
+  @Override
+  void doMergeAndReset(Aggregator target) {
+    DoubleSummaryAggregator other = (DoubleSummaryAggregator) target;
+
+    DoubleSummary copy = current.copyAndReset();
+    other.current.update(copy.count, copy.sum, copy.min, copy.max);
```

Comment:
I was thinking of doing it just because it reads better. I don't know if it actually saves any allocations anywhere. I'll do it for cleanliness, though.  :)

Reply:
done

Reply:
you don't need to create the DoubleSummary copy, so it looks to me like an allocation save.